patches/revert_nvcomp4.patch

```diff
@@ -1,907 +1,140 @@
--  return temp_size;
-+// Dispatcher for nvcompBatched<format>DecompressGetTempSize
-+template <typename... Args>
-+auto batched_decompress_get_temp_size(compression_type compression, Args&&... args)
-+{
-+  switch (compression) {
-+    case compression_type::SNAPPY:
-+      return nvcompBatchedSnappyDecompressGetTempSize(std::forward<Args>(args)...);
-+    case compression_type::ZSTD:
-+#if NVCOMP_HAS_ZSTD_DECOMP(NVCOMP_MAJOR_VERSION, NVCOMP_MINOR_VERSION, NVCOMP_PATCH_VERSION)
-+      return nvcompBatchedZstdDecompressGetTempSize(std::forward<Args>(args)...);
-+#else
-+      CUDF_FAIL("Decompression error: " +
-+                nvcomp::is_decompression_disabled(nvcomp::compression_type::ZSTD).value());
-+#endif
-+    case compression_type::DEFLATE:
-+#if NVCOMP_HAS_DEFLATE(NVCOMP_MAJOR_VERSION, NVCOMP_MINOR_VERSION, NVCOMP_PATCH_VERSION)
-+      return nvcompBatchedDeflateDecompressGetTempSize(std::forward<Args>(args)...);
-+#else
-+      CUDF_FAIL("Decompression error: " +
-+                nvcomp::is_decompression_disabled(nvcomp::compression_type::DEFLATE).value());
-+#endif
-+    case compression_type::LZ4:
-+      return nvcompBatchedLZ4DecompressGetTempSize(std::forward<Args>(args)...);
-+    default: CUDF_FAIL("Unsupported compression type");
-+  }
- }
- 
- // Dispatcher for nvcompBatched<format>DecompressAsync
-@@ -72,12 +120,20 @@ auto batched_decompress_async(compression_type compression, Args&&... args)
-     case compression_type::SNAPPY:
-       return nvcompBatchedSnappyDecompressAsync(std::forward<Args>(args)...);
-     case compression_type::ZSTD:
-+#if NVCOMP_HAS_ZSTD_DECOMP(NVCOMP_MAJOR_VERSION, NVCOMP_MINOR_VERSION, NVCOMP_PATCH_VERSION)
-       return nvcompBatchedZstdDecompressAsync(std::forward<Args>(args)...);
-+#else
-+      CUDF_FAIL("Decompression error: " +
-+                nvcomp::is_decompression_disabled(nvcomp::compression_type::ZSTD).value());
-+#endif
-     case compression_type::DEFLATE:
-+#if NVCOMP_HAS_DEFLATE(NVCOMP_MAJOR_VERSION, NVCOMP_MINOR_VERSION, NVCOMP_PATCH_VERSION)
-       return nvcompBatchedDeflateDecompressAsync(std::forward<Args>(args)...);
-+#else
-+      CUDF_FAIL("Decompression error: " +
-+                nvcomp::is_decompression_disabled(nvcomp::compression_type::DEFLATE).value());
-+#endif
-     case compression_type::LZ4: return nvcompBatchedLZ4DecompressAsync(std::forward<Args>(args)...);
--    case compression_type::GZIP:
--      return nvcompBatchedGzipDecompressAsync(std::forward<Args>(args)...);
-     default: CUDF_FAIL("Unsupported compression type");
-   }
- }
-@@ -89,11 +145,31 @@ std::string compression_type_name(compression_type compression)
-     case compression_type::ZSTD: return "Zstandard";
-     case compression_type::DEFLATE: return "Deflate";
-     case compression_type::LZ4: return "LZ4";
--    case compression_type::GZIP: return "GZIP";
+diff --git a/cpp/src/strings/regex/regcomp.cpp b/cpp/src/strings/regex/regcomp.cpp
```

Comment:
Curious why we're patching regcomp and string extract files as part of fixing cudf to have nvcomp 3.x?

Reply:
Since 24.12 build is totally busted, filed #2498 to track this issue.  It's pretty nasty, since it silently doesn't build any changes.  Running with outdated binaries relative to the source after a successful build is confusing to debug.